src/content/docs/beyond-splashkit/Maths/1-math-basics.mdx

```diff
@@ -0,0 +1,2081 @@
+### Finding Prime Numbers up to a Given Number
+
+Finding prime numbers up to a given number is a common task in number theory and cryptography. The following function implements the Sieve of Eratosthenes algorithm, which efficiently finds all prime numbers up to a specified maximum $n$. The algorithm works by:
+
+1. Creating a list of numbers from 2 to $n$
+2. Starting with $p=2$ (the first prime), marking all multiples of $p$ up to $n$ as composite
+3. Finding the next unmarked number and repeating step 2
+4. The remaining unmarked numbers are prime
+
+This can be expressed mathematically as finding all numbers $p$ where:
+
+$$
+p \in \{2,\ldots,n\} \text{ and } p \nmid k \text{ for all } k < p
+$$
+
+For example, to find primes up to 20, we would mark off:
+
+- Multiples of 2: $4, 6, 8, 10, 12, 14, 16, 18, 20$
+- Multiples of 3: $9, 15$
+- Multiples of 5: none remaining
+- Leaving primes: $2, 3, 5, 7, 11, 13, 17, 19$
+
+:::note
+This function relies on the `is_the_number_prime` function defined above.
+:::
+
+<Tabs syncKey="code-language">
+<TabItem label="C++">
+
+```cpp
+void find_prime_numbers_up_to(int limit)
+{
+    std::cout << "Prime numbers up to " << limit << ":";
+
+    // Check all numbers up to the limit to see if they are prime
+    for (int i = 2; i <= limit; i++)
+    {
+        if (is_the_number_prime(i))
+        {
+            std::cout << i << " ";
+        }
+    }
+    std::cout << std::endl;
+}
+```
+
+</TabItem>
+<TabItem label="C#">
+
+```csharp
+void FindPrimeNumbersUpTo(int limit)
+{
+    Console.WriteLine($"Prime numbers up to {limit}:");
+
+    // Check all numbers up to the limit to see if they are prime
+    for (int i = 2; i <= limit; i++)
+    {
+        if (IsTheNumberPrime(i))
+        {
+            Console.Write($"{i} ");
+        }
+    }
+    Console.WriteLine("");
+}
+```
+
+</TabItem>
+<TabItem label="Python">
+
+```python
+def find_prime_numbers_up_to(limit):
+    print(f"Prime numbers up to {limit}:")
+
+    # Check all numbers up to the limit to see if they are prime
+    for i in range(2, limit + 1):
+        if is_the_number_prime(i):
+            print(f"{i} ")
+    print("")
+```
+
+</TabItem>
+</Tabs>
+
+:::tip
+This could be useful for understanding number theory, building encryption algorithms, or solving problems in combinatorics.
+:::
+
+#### Using this function
+
+You can use the `find_prime_numbers_up_to` function to find all prime numbers up to a given limit. For example, to find prime numbers up to $40$:
+
+<Tabs syncKey="code-language">
+<TabItem label="C++">
+
+```cpp
+find_prime_numbers_up_to(40);
+```
+
+</TabItem>
+<TabItem label="C#">
+
+```csharp
+FindPrimeNumbersUpTo(40);
+```
+
+</TabItem>
+<TabItem label="Python">
+
+```python
+find_prime_numbers_up_to(40)
+```
+
+</TabItem>
+</Tabs>
+
+#### Expected Output
+
+After running the above code, the expected output should be:
+
+```plaintext
+Prime numbers up to 40:
+2 3 5 7 11 13 17 19 23 29 31 37
+```
+
+### Eulers Totient Function
```

Comment:
I would suggest Euler's Totient Function be moved down, at least after the explanation of coprimes. As this is a beginner guide they may not know of coprimes and they are explained further down